lib/src/json_schema/json_schema.dart

```diff
@@ -1962,4 +1997,17 @@ class JsonSchema {
 
     this._validateAndSetAllProperties();
   }
+
+  // If this is a recursive Anchor, find it's first recursive anchor parent.
+  JsonSchema findRecursiveParent() {
+    var curHit = this;
+    var curNode = this;
+    while (curNode._parent != null) {
+      curNode = curNode._parent;
+      if (curNode.recursiveAnchor) {
+        curHit = curNode;
+      }
+    }
+    return curHit;
+  }
```

Comment:
This code doesn't seem to check whether the first `curNode` (`this`) has a recursive anchor. Maybe the calling code does that check before calling this function? But then the function docstring isn't quite right.

Reply:
Something like this makes sense to me. Am I missing something?
```dart
  JsonSchema resolveAnchored() {
    var curNode = this;
    while (curNode.recursiveAnchor && curNode._parent != null) {
      curNode = curNode._parent;
    }
    return curNode;
  }
```

Reply:
oh, that doc string is wrong now. The function is finding the furthest away parent that has a `recursiveAnchor` There will be parents in between that don't have `recursiveAnchor` set.